eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/mutable/UnmodifiableBag.java

```diff
@@ -64,7 +64,7 @@
         extends AbstractUnmodifiableMutableCollection<T>
         implements MutableBag<T>, Serializable
 {
-    UnmodifiableBag(MutableBag<? extends T> mutableBag)
+    protected UnmodifiableBag(MutableBag<? extends T> mutableBag)
```

Comment:
Note how UnmodifiableBag extends AbstractUnmodifiableMutableCollection. I think the right thing to do here is extract a new superclass AbstractUnmodifiableBag, which becomes part of our public api. I think we should leave the sublcass alone.

Reply:
what would be left in the `UnmodifiableBag` class at then end? nothing except being final for security purpose?

Reply:
That's right. I'm not even sure I know why, but I see that the JDK does it and Guava does it, and I think we must have done it for a similar reason.